File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0)  return 0;
        int cnt = 1;
        int maxi = 0;
        sort(nums.begin(), nums.end());

        for(int i=1;i<nums.size();i++)
        {
            if(nums[i] == nums[i-1])  continue;

            if(nums[i] - nums[i-1] == 1)
            {
                cnt++;
                
            }
            else {
                maxi = max(maxi, cnt);
                cnt = 1;
            }

            
        }
        return max(maxi, cnt);
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        Arrays.sort(nums);

        int cnt = 1;
        int maxi = 0;

        for (int i = 1; i < nums.length; i++) {

            // Skip duplicates
            if (nums[i] == nums[i - 1]) {
                continue;
            }

            // Consecutive element
            if (nums[i] - nums[i - 1] == 1) {
                cnt++;
            } 
            // Reset count
            else {
                maxi = Math.max(maxi, cnt);
                cnt = 1;
            }
        }

        return Math.max(maxi, cnt);
    }
}